cache/memcached_client.go

```diff
@@ -178,15 +179,21 @@ func newMemcachedClient(
 	reg prometheus.Registerer,
 	name string,
 ) (*memcachedClient, error) {
+	newRegisterer := prometheus.WrapRegistererWith(
```

Comment:
I would do the naming the other way round. I would call `newRegisterer` just `reg` and the other `legacyReg` to clarify which one is the legacy one.